x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java

```diff
@@ -454,8 +444,13 @@ public boolean functionExists(String functionName) {
 
     @SuppressWarnings("overloads")
     private static FunctionDefinition def(Class<? extends Function> function, FunctionBuilder builder,
-            boolean datetime, String... aliases) {
-        String primaryName = normalize(function.getSimpleName());
+            boolean datetime, String... names) {
+        Check.isTrue(names.length > 0, "At least one name must be provided for the function");
+        String primaryName = names[0];
+        List<String> aliases = new ArrayList<>(names.length - 1);
```

Comment:
I would add a comment here or a javadoc for the method stating that the first name in the array is the actual name of the function and the rest (if any) are aliases.